### ISSUE TITLE
ServerlessSpec is not optional and fails validation with schema

### ISSUE THREAD

**Issue by Sufian22** [mongodb/helm-charts#364]
**What did you do to encounter the bug?**
1. Generate the K8s template files (with `helm template`) for [mongodb/atlas-deployment](https://github.com/mongodb/helm-charts/tree/main/charts/atlas-deployment) helm chart 

**What did you expect?**
I expected no `serverlessSpec` field in the AtlasDeployment if none has been specified in the `values.yaml` file.

**What happened instead?**
It generates a `spec.serverlessSpec` with `null` value inside the `AtlasDeployment` which doesn't satisfy the schema definition.

**Screenshots**
The `if` condition should be added also to the `serverlessSpec` too:
https://github.com/mongodb/helm-charts/blob/4a5a00f7869f77234fe22972a69c3ad2847bc834/charts/atlas-deployment/templates/atlas-deployment.yaml#L23

**Kubernetes Cluster Information**
 - Distribution
 - Version
 - Image Registry location (quay, or an internal registry)

**Additional context**
- Atlas-Operator and Atlas-Deployment helm charts both in version `2.4.1`

**Comment by josvazg** [comment]
Thanks for the report @Sufian22 , spot on!
@igor-karpukhin has already posted #365 to solve this

**Comment by Sufian22** [comment]
@josvazg I'm finding some incompatibilities with the generated `AtlasDeployment` resource and it's schema definition, do you want me to create another issue for that? 🤔

**Comment by josvazg** [comment]
> @josvazg I'm finding more incompatibilities with the generated `AtlasDeployment` resource and it's schema definition, do you want me to create another issue for that? 🤔

Yes, please. If the issues are not related to this same exact fix, it is better to have a separate issue.
